Calculator.js

```diff
@@ -1,6 +1,184 @@
+  },
+
+  shouldDisplay: function () {
+    return this.displayFlag
+  },
+
+  // Special operation functions
+
+  allClear: function () {
+    this.operands = []
+    this.operations = []
+    this.display = '0'
+    this.displayFlag = true
+    this.value = '0'
+    return this.displayScreen()
+  },
+
+  changeSign: function () {
+    if ( !this.checkForSignInDisplay() ) {
+      this.display = '-' + this.display
+    } else {
+      this.display = this.display.slice( -( this.getDisplayLength() - 1 ) )
+    }
+  },
+
+  computePercentage: function () {
+    this.display = ( this.display / 100 )
+    this.value = ''
+    return this.displayScreen()
+  },
+
+  completeOperations: function () {
+    while ( this.operations.length ) {
+      this.display = this.lookUpFunctions(
+        this.operations.pop(),
+        this.display,
+        this.operands.pop()
+      )
+    }
+    this.value = ''
+    this.displayFlag = true
+    return this.displayScreen()
+  },
+
+  // Functions that control calculator based on input
+
+  handleInput: function () {
+    if ( this.isNumber() ) {
+      this.displayScreen()
+      return
+    }
+    if ( this.isOperation() ) {
+      this.handleOperations()
+      return
+    }
+    return this.handleSpecialOperations()
+  },
+
+  handleOperations: function () {
+    if ( this.isEqualSign() ) {
+      return this.completeOperations()
+    }
+    if ( !this.queuedOperations() ) {
+      this.operations.push( this.value )
+      this.operands.push( this.display )
+      this.displayFlag = false
+      this.display = '0'
+    }
+  },
+
+  handleSpecialOperations: function () {
+    if ( this.checkForAllClear() ) {
+      return this.allClear()
+    }
+    if ( this.checkForDecimal() ) {
+      if ( !this.displayContainsDecimal() ) {
+        this.displayScreen()
+      }
+      return
+    }
+    if ( this.checkForSign() ) {
+      return this.changeSign()
+    }
+    if ( this.checkForPercentage() ) {
+      this.computePercentage()
+    }
+  },
+
+  // Boolean functions
+
+  checkForAllClear: function () {
+    return this.value === 'AC'
+  },
+
+  checkForDecimal: function () {
+    return this.value === '.'
+  },
+
+  checkForPercentage: function () {
+    return this.value === '%'
+  },
+
+  checkForSign: function () {
+    return this.value === 'sign'
+  },
+
+  checkForSignInDisplay: function () {
+    return this.display.includes( '-' )
+  },
+
+  displayContainsDecimal: function () {
+    return this.display.includes( '.' )
+  },
+
+  isEqualSign: function () {
+    return this.value === '='
+  },
+
+  isNumber: function () {
+    return [ 0, 1, 2, 3, 4, 5, 6, 7, 8, 9 ].includes( parseInt( this.value ) )
+  },
+
+  isOperation: function () {
+    return [ '=', 'X', '/', '+', '-' ].includes( this.value )
+  },
+
+  queuedOperations: function () {
+    return this.operations.length !== 0
+  },
+
+  // Look up table for functions
+
+  lookUpFunctions: function ( operator, operandOne, operandTwo ) {
```

Comment:
one of the specs requires that each mathematical operation be stored as a discrete and pure unction. Break this up into functions.